framework/doc/content/source/meshgenerators/CoarsenBlockGenerator.md

```diff
@@ -0,0 +1,41 @@
+# CoarsenBlockGenerator
+
+!syntax description /Mesh/CoarsenBlockGenerator
+
+## Overview
+
+This MeshGenerator object allows the user to coarsen one or more blocks in a mesh. It has been designed
+to work with previously uniformly refined meshes, though the heuristic can still work in the general case.
```

Comment:
This documentation combined with what is happening in the source code really confused me. Based on this documentation I would have expected that this generator only works if the mesh has been refined, e.g. that element levels are > 0. But instead it is the opposite
```c++
      if (current_elem->level() > 0)
        mooseError("H-refined meshes cannot be coarsened with this mesh generator. Use the "
                   "[Adaptivity] block to coarsen them.");
```
It would be a nice symmetry if `CoarsenBlockGenerator` reused the code from `RefineBlockGenerator` but instead set `Elem::COARSEN` instead of `Elem::REFINE`.

Of course that will be a different functionality than what you are actually doing in your code here. You'll need to change these sentences to not confuse people. I might even suggest a different name for the class although I could be persuaded otherwise from a user perspective. From a libMesh developer perspective, I have a hard time not expecting `CoarsenBlockGenerator` to be the coarsening analog, and potentially reusing the code, of `RefineBlockGenerator`

Reply:
> It would be a nice symmetry if CoarsenBlockGenerator reused the code from RefineBlockGenerator but instead set Elem::COARSEN instead of Elem::REFINE.

sure but it also would not be very useful since we could only work with meshes refined with libmesh AMR (and that 'need' is fully covered with initial adaptivity steps)

I ll try to modify the sentence